### ISSUE TITLE
Access to the groud truth during training

### ISSUE THREAD

**Issue by hui-po-wang** [lunanbit/UUlearning#1]
Hi,

Thank you for sharing code.

To my knowledge, it seems that the major contribution of your paper is to train a binary classifier with only the access to the theta and theta'. That is, you won't know any label of a single sample. 

If so, I got confused after reading your code. https://github.com/lunanbit/UUlearning/blob/master/loss.py#L29 
It seems that you still access labels and use them to weight your loss?

If I made any mistake, please correct me.

--------------------------- Edit -------------------------------

Or, do the pos and neg in the code represent which unlabeled datasets U they are drawn from?

**Comment by lunanbit** [comment]
Hi,

Thanks for your interest in our work!

The main contribution of this paper is to train a binary classifier with only access to theta, theta' and pip from unlabeled training data. The proposed method actually treats the U set with the larger/smaller class prior as the corrupted positive/negative dataset and then use the loss correction technique to make the corrected loss perfectly compatible with the corrupted data.

In the code, we have NO access to any label of a single training example. The pos and neg in the code only indicates from which U set the sample is drawn, since the U set with the larger/smaller class prior is regarded as the corrupted P/N dataset.

Hope this helps to clarify your question^^
